pkg/image/fetcher.go

```diff
@@ -171,8 +178,28 @@ func (f *Fetcher) fetchDaemonImage(name string) (imgutil.Image, error) {
 	return image, nil
 }
 
-func (f *Fetcher) fetchRemoteImage(name string) (imgutil.Image, error) {
-	image, err := remote.NewImage(name, f.keychain, remote.FromBaseImage(name))
+func (f *Fetcher) fetchRemoteImage(name string, target *dist.Target) (imgutil.Image, error) {
+	var (
+		image imgutil.Image
+		err   error
+	)
+
+	if target == nil {
+		image, err = remote.NewImage(name, f.keychain, remote.FromBaseImage(name))
+	} else {
+		platform := imgutil.Platform{OS: target.OS, Architecture: target.Arch}
+		// when targeting a registry, we need to use variant if available to hit the correct image
+		platform.Variant = target.ArchVariant
+		if len(target.Distributions) > 0 {
+			// We assumed the given target's distributions were expanded, and we are just dealing with 1 version.
+			platform.OSVersion = target.Distributions[0].Version
+			if len(target.Distributions) > 1 {
+				f.logger.Warnf("trying to fetch an image with more than one OS distribution, using %s", style.Symbol(target.Distributions[0].Version))
+			}
+		}
```

Comment:
Do we still need this part? I think we determined that distribution information is ignored when fetching?

Reply:
I think similar to my previous comment, we don't need it for fetching but we need it for creating the image with the given version in the config file. I added your suggested comment

Reply:
But, this is just fetching?

Reply:
I forgot to remove the warning, we don't need the warning, but we need the osVersion to configure the config file on [imgutil](https://github.com/buildpacks/imgutil/blob/main/remote/new.go#L82-L87)

Reply:
We confirmed the fetch call isn't used to create new (blank) images, so we removed the distribution stuff